src/components/Api.js

```diff
@@ -0,0 +1,98 @@
+export default class Api {
+  constructor({ headers, baseUrl }) {
+    this._headers = headers;
+    this._baseUrl = baseUrl;
+  }
+
+  _checkResponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    return Promise.reject(`Error: ${res.status}`);
+  }
+
+  getUserInfo() {
+    return fetch(`${this._baseUrl}/users/me`, {
+      headers: this._headers
+    }).then(this._checkResponse)
+    .then((res) => {console.log(res); return res;})
+    .catch((err) => {
+      console.log(err);
```

Comment:
It’s better not to use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request and this end is in `index.js`  after all blocks `then`. And in real projects blocks `catch` usually open error popups and so on.

Reply:
Understood. This has been removed now